content_harvester/by_page.py

```diff
@@ -0,0 +1,397 @@
+import json
+import settings
+import boto3
+import os
+import requests
+import derivatives
+from botocore.exceptions import ClientError
+from requests.adapters import HTTPAdapter, Retry
+from collections import Counter
+
+
+class DownloadError(Exception):
+    pass
+
+
+def get_mapped_records(collection_id, page_filename, s3_client) -> list:
+    mapped_records = []
+    if settings.METADATA_SRC == 'local':
+        local_path = settings.local_path(
+            'mapped_metadata', collection_id)
+        page_path = os.sep.join([local_path, str(page_filename)])
+        page = open(page_path, "r")
+        mapped_records = json.loads(page.read())
+    else:
+        page = s3_client.get_object(
+            Bucket=settings.S3_BUCKET,
+            Key=f"mapped_metadata/{collection_id}/{page_filename}"
+        )
+        mapped_records = json.loads(page['Body'].read())
+    return mapped_records
+
+
+def write_mapped_record(collection_id, record, s3_client):
+    if settings.METADATA_DEST == 'local':
+        local_path = settings.local_path(
+            'mapped_with_content', collection_id)
+        page_path = os.sep.join([local_path, str(record.get('calisphere-id'))])
+        page = open(page_path, "w")
+        page.write(json.dumps(record))
+    else:
+        upload_status = s3_client.put_object(
+            Bucket=settings.S3_BUCKET,
+            Key=f"mapped_with_content/{collection_id}/{record.get('calisphere-id')}",
+            Body=json.dumps(record)
+        )
+        print(f"Upload status: {upload_status}")
+
+
+def get_child_records(collection_id, parent_id, s3_client) -> list:
+    mapped_child_records = []
+    if settings.METADATA_SRC == 'local':
+        local_path = settings.local_path('mapped_metadata', collection_id)
+        children_path = os.sep.join([local_path, 'children'])
```

Comment:
Why not use `os.path.join` here? From my understanding after having read about `os.sep.join` vs `os.path.join`, the latter is more intelligent. You could end up with 2 separators in a row when using `os.sep.join`.

See https://docs.python.org/3/library/os.html?highlight=os%20sep#os.sep